selfdrive/car/toyota/carcontroller.py

```diff
@@ -128,20 +128,27 @@ def update(self, enabled, CS, frame, actuators, pcm_cancel_cmd, hud_alert,
 
     # steer torque
     new_steer = int(round(actuators.steer * SteerLimitParams.STEER_MAX))
-    apply_steer = apply_toyota_steer_torque_limits(new_steer, self.last_steer, CS.steer_torque_motor, SteerLimitParams)
-    self.steer_rate_limited = new_steer != apply_steer
 
     # only cut torque when steer state is a known fault
     if CS.steer_state in [9, 25]:
       self.last_fault_frame = frame
 
     # Cut steering for 2s after fault
-    if not enabled or (frame - self.last_fault_frame < 200):
-      apply_steer = 0
+    if (frame - self.last_fault_frame < 200) or \
+       (abs(CS.angle_steers_rate) > 100 and CS.CP.carFingerprint in (CAR.RAV4H, CAR.COROLLA)) or \
+       (CS.CP.carFingerprint == CAR.RAV4H and abs(CS.angle_steers) > 100):
+      new_steer = 0
       apply_steer_req = 0
```

Comment:
Do you want the steer_request bit to be zero while you're winding down?

Reply:
I have tried it without switching it off and then there are still some occurrences where the steering faults. Having it switch to 0 steering request is the right solution.